Examples/Modules/embedded_boundary_cube/analysis_fields.py

```diff
@@ -0,0 +1,99 @@
+ncells = [48, 48, 48]
+dx = (hi[0] - lo[0])/ncells[0]
+dy = (hi[1] - lo[1])/ncells[1]
+dz = (hi[2] - lo[2])/ncells[2]
+m = 0
+n = 1
+p = 1
+Lx = 1
+Ly = 1
+Lz = 1
+h_2 = (m * pi / Lx) ** 2 + (n * pi / Ly) ** 2 + (p * pi / Lz) ** 2
+t = 1.3342563807926085e-08
+
+# Compute the analytic solution
+Bx_th = np.zeros(ncells)
+By_th = np.zeros(ncells)
+Bz_th = np.zeros(ncells)
+for i in range(ncells[0]):
+    for j in range(ncells[1]):
+        for k in range(ncells[2]):
+            x = i*dx + lo[0]
+            y = j*dy + lo[1]
+            z = k*dz + lo[2]
+
+            Bx_th[i, j, k] = -2/h_2*mu_0*(m * pi/Lx)*(p * pi/Lz) * (np.sin(m * pi/Lx * (x - Lx/2)) *
+                                                                    np.sin(n * pi/Ly * (y - Ly/2)) *
+                                                                    np.cos(p * pi/Lz * (z - Lz/2)) *
+                                                                    (-Lx/2 < x < Lx/2) *
+                                                                    (-Ly/2 < y < Ly/2) *
+                                                                    (-Lz/2 < z < Lz/2) *
+                                                                    np.cos(np.sqrt(2) *
+                                                                           np.pi / Lx * c * t))
+
+            By_th[i, j, k] = -2/h_2*mu_0*(n * pi/Ly)*(p * pi/Lz) * (np.cos(m * pi/Lx * (x - Lx/2)) *
+                                                                    np.sin(n * pi/Ly * (y - Ly/2)) *
+                                                                    np.cos(p * pi/Lz * (z - Lz/2)) *
+                                                                    (-Lx/2 < x < Lx/2) *
+                                                                    (-Ly/2 < y < Ly/2) *
+                                                                    (-Lz/2 < z < Lz/2) *
+                                                                    np.cos(np.sqrt(2) *
+                                                                    np.pi / Lx * c * t))
+
+            Bz_th[i, j, k] = mu_0*(n * pi/Ly)*(p * pi/Lz) * (np.cos(m * pi/Lx * (x - Lx/2)) *
+                                                             np.cos(n * pi/Ly * (y - Ly/2)) *
+                                                             np.sin(p * pi/Lz * (z - Lz/2)) *
+                                                             (-Lx/2 < x < Lx/2) *
+                                                             (-Ly/2 < y < Ly/2) *
+                                                             (-Lz/2 < z < Lz/2) *
+                                                             np.cos(np.sqrt(2) *
+                                                                    np.pi / Lx * c * t))
+
+
+# Open the right plot file
+filename = sys.argv[1]
+ds = yt.load(filename)
+data = ds.covering_grid(level=0, left_edge=ds.domain_left_edge, dims=ds.domain_dimensions)
+
+tol_err = 1e-5
```

Comment:
My understanding is that this value is probably too high for this test to properly check the simulated fields.

More specifically, if I didn't make any mistake, I think even if `Bx_sim`/`By_sim`/`Bz_sim` were completely wrong (e.g. `Bx_sim = By_sim = Bz_sim = 0`), then the `err_x`/`err_y`/`err_z` would be on the order of `1.e-7` or `1.e-6`, i.e. below `tol_err`, which means that the test would pass.

One could use a more stringent value for `tol_err`. 
Or one alternative maybe is to use a relative error like this:
```
rel_tol_err = 0.1
Bx_sim = data['Bx'].to_ndarray()
rel_err_x = np.sqrt( np.sum(np.square(Bx_sim - Bx_th)) / np.sum(np.square(Bx_th)) )
assert(rel_err_x < rel_tol_err )
```

Reply:
Good point! The relative error you are suggesting seems good to me, I will implement it. Thanks a lot for pointing out this.

Reply:
I have implemented the relative error. By the way, I am not checking the error on Bx anymore as it is zero everywhere and the relative error would blow up.